drivers/spi/Kconfig.nrfx

```diff
@@ -35,18 +35,17 @@ config SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58
 if SPI_0 && (SOC_NRF52810 || \
              (!I2C_0 && !(SOC_SERIES_NRF91X && UART_0_NRF_UARTE)))
 
-choice
-	prompt "SPI Port 0 Driver type"
-
 config SPI_0_NRF_SPI
-	bool "nRF SPI 0"
+	def_bool y
+	depends on $(dt_int_val,DT_NORDIC_NRF_SPI_SPI_0_BASE_ADDRESS) != 0
```

Comment:
If you want to simplify it some more, you could do

```
def_bool $(dt_int_val,DT_NORDIC_NRF_SPI_SPI_0_BASE_ADDRESS) != 0
```

and get rid of the `depends on` (and ditto for the other ones).

Any kind of expression can appear after `default`, `def_bool`, etc.

Reply:
Will do. Thanks for the hint.